qpmodel/PhysicalNode.cs

```diff
@@ -1529,61 +1543,37 @@ protected override double EstimateCost()
 
     public class PhysicFromQuery : PhysicNode
     {
-        List<Row> cteCache_ = null;
+        public bool setAsInlineCte() => IsInlineCte = true;
+        public bool IsInlineCte { get; private set; } = false;
 
         public override string ToString() => $"PFrom({(logic_ as LogicFromQuery)}: {Cost()})";
 
         public PhysicFromQuery(LogicFromQuery logic, PhysicNode l) : base(logic) => children_.Add(l);
 
-        public bool IsCteConsumer(out CTEQueryRef qref) => (logic_ as LogicFromQuery).IsCteConsumer(out qref);
+        public PhysicFromQuery(LogicCteConsumer logic, PhysicNode l) : base(logic) => children_.Add(l);
 
         public override void Open(ExecContext context)
         {
             base.Open(context);
-            var logic = logic_ as LogicFromQuery;
-            if (logic.IsCteConsumer(out CTEQueryRef qref))
```

Comment:
So with a sequence plan, a cteProducer generates a query result, later cteConsumers can reuse the result - it is used to be in cteCache_, where is that logic now?

Reply:
cte name will be registerd in CTEProducer when Bind CTEProducer.  and cte consumer can read from the context.